includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -461,8 +461,30 @@ private function get_used_class_names() {
 			$classes .= ' ' . $class_attribute->nodeValue;
 		}
 
-		// Find all [class] attributes and capture the contents of any single- or double-quoted strings.
+		// Find all [class] attributes and capture the contents of any single- or double-quoted strings,
+		// as well as <amp-state> content that might be used.
 		foreach ( $this->xpath->query( '//*/@' . AMP_DOM_Utils::AMP_BIND_DATA_ATTR_PREFIX . 'class' ) as $bound_class_attribute ) {
+			if ( preg_match_all( '/([a-zA-Z0-9_]+?)\[\s*([\'"])([^\2]*?)\2\s*\]/', $bound_class_attribute->nodeValue, $matches ) ) {
```

Comment:
Please provide some examples above for what this pattern will be matching.

Reply:
This was removed as not needed anymore.